PR: https://github.com/odoo/odoo/pull/215789

From: 9a7d9ae8a3328d6284c8a62538a4a2df16b1873f
From: alan-odoo
Date: 2025-09-06 21:40:47

Breaking data model changes scores: del:3 + add:0, change matches:
-    attendance_state = fields.Selection(related='employee_id.attendance_state')
-    last_check_in = fields.Datetime(related='employee_id.last_attendance_id.check_in')
-    last_check_out = fields.Datetime(related='employee_id.last_attendance_id.check_out')

Total Changes: 6

[REM] *: remove unused fields of res.users

*=hr_attendance, hr_holidays

This commit cleans the res.users model by removing its unused fields.

Enterprise PR: https://github.com/odoo/enterprise/pull/88518
Upgrade PR: https://github.com/odoo/upgrade/pull/8207

Task-4775990

Part-of: odoo/odoo#215789
Related: odoo/enterprise#88518
Related: odoo/upgrade#8207
Signed-off-by: Thibault Delavallee (tde) <tde@openerp.com>

================================= pseudo patch: =================================

--- a/addons/hr_attendance/models/res_users.py
+++ b/addons/hr_attendance/models/res_users.py
@@ -9,9 +9,6 @@ class ResUsers(models.Model):
 
     hours_last_month = fields.Float(related='employee_id.hours_last_month')
     hours_last_month_display = fields.Char(related='employee_id.hours_last_month_display')
-    attendance_state = fields.Selection(related='employee_id.attendance_state')
-    last_check_in = fields.Datetime(related='employee_id.last_attendance_id.check_in')
-    last_check_out = fields.Datetime(related='employee_id.last_attendance_id.check_out')
     total_overtime = fields.Float(related='employee_id.total_overtime')
     hours_last_month_overtime = fields.Float(related='employee_id.hours_last_month_overtime')
     attendance_manager_id = fields.Many2one(related='employee_id.attendance_manager_id', readonly=False)
@@ -22,9 +19,6 @@ class ResUsers(models.Model):
         return super().SELF_READABLE_FIELDS + [
             'hours_last_month',
             'hours_last_month_display',
-            'attendance_state',
-            'last_check_in',
-            'last_check_out',
             'total_overtime',
             'hours_last_month_overtime',
             'attendance_manager_id',
